### PR TITLE
Update deps GH workflow 

### PR DESCRIPTION
**Pull request by F-WRunTime**

### CODE REVIEW

File: .github/workflows/master-push.yml [hunk 1]
```diff
@@ -53,3 +53,17 @@ jobs:
...
+
+      - name: 'Update dependents'
+        env:
+          GITHUB_TOKEN: ${{ secrets.JENKINS_GITHUB_PAT }}
+        run: |
+          set -x
+          version="${GITHUB_SHA}"
+          curl --fail                                                          \
+            -X POST                                                            \
+            -H "Accept: application/vnd.github+json"                           \
```
Comment (ehildenb) [inline .github/workflows/master-push.yml]: This should run in the `release.yml` workflow, and replace this: https://github.com/runtimeverification/k/blob/master/.github/workflows/release.yml#L345

Note there that we also are not using `GITHUB_SHA` for the version, but the output from a previous stage.

Comment (ehildenb) [reply .github/workflows/master-push.yml]: This should be removed now.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -355,17 +355,20 @@ jobs:
...
           VERSION: ${{ needs.macos-build.outputs.version }}
         run: |
-          curl --fail 'https://ci.runtimeverification.com/jenkins/buildByToken/buildWithParameters' \
-              --data job=Devops/master \
-              --data token=$JENKINS_DEVOPS_TOKEN \
-              --data UPDATE_DEPS=true \
-              --data UPDATE_DEPS_REPO=runtimeverification/k \
-              --data UPDATE_DEPS_VERSION=v$VERSION
+          set -x
+          version="${GITHUB_SHA}"
```
Comment (ehildenb) [inline .github/workflows/release.yml]:
```suggestion
          set -x
          version="${VERSION}"
```

Comment (ehildenb) [reply .github/workflows/release.yml]: You can also remove the `v` from eth line 371 in `"version":"'v${VERSION}'"`, and add it here instead, if you want.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -355,17 +355,20 @@ jobs:
...
           VERSION: ${{ needs.macos-build.outputs.version }}
         run: |
-          curl --fail 'https://ci.runtimeverification.com/jenkins/buildByToken/buildWithParameters' \
-              --data job=Devops/master \
-              --data token=$JENKINS_DEVOPS_TOKEN \
-              --data UPDATE_DEPS=true \
-              --data UPDATE_DEPS_REPO=runtimeverification/k \
-              --data UPDATE_DEPS_VERSION=v$VERSION
+          set -x
+          version=v"${GITHUB_SHA}"
```
Comment (ehildenb) [inline .github/workflows/release.yml]:
```suggestion
          version=v"${VERSION}"
```

**Comment by F-WRunTime** [comment]
Thanks copy / paste error there

**Review by ehildenb — APPROVED** [review]

**Pull request merged**